src/client.rs

```diff
@@ -2020,6 +2104,146 @@ mod tests {
         assert_eq!(ids, [1, 2, 3, 4, 5].iter().cloned().collect::<Vec<_>>());
     }
 
+    #[test]
```

Comment:
I can implement some integration tests,  The issue I see right now is I can build tests but only can test them on a real server (cyrus imap in my case). as Greenmail does not currently support the ACL extension. (they have comments in their code for where it would be added).   I've added a commit ( 85f8d0c6932cac714843fc5cce1b7da4df78b18a ) to pull out the host/port methods so that I can override ALL via ENV to allow me to more easily test that.  ( I can also provide a Dockerfile to setup a cyrus imap setup that will work like greenmail once I get the SMTP part done)

Reply:
Ahh, that's a shame. If you have a test that works locally against a real server, then I'm okay merging with that test not being run against Greenmail :+1: 

Having integration test setup against Cyrus as well as Greenmail would be amazing! The fact that we currently only test against a single server implementation is certainly a shortcoming.

Reply:
@jonhoo are you OK with skipping the integration tests in this PR and focusing on building those in #228 ?

Reply:
I think maybe we should try to land #228 so that we can then add the tests directly in this PR. What do you think?